src/main/java/uk/gov/companieshouse/web/accounts/transformer/smallfull/impl/CreditorsAfterOneYearTransformerImpl.java

```diff
@@ -183,12 +180,19 @@ private void setPreviousPeriodOnApiModel(CreditorsAfterOneYear creditorsAfterOne
         }
     }
 
-    private boolean isPreviousPeriodPopulated(PreviousPeriod period) {
-        return Stream.of(
-                period.getBankLoansAndOverdrafts(),
-                period.getFinanceLeasesAndHirePurchaseContracts(),
-                period.getOtherCreditors(),
-           period.getTotal()).anyMatch(Objects ::nonNull);
+    private boolean isCurrentPeriodPopulated(CurrentPeriod currentPeriod) {
```

Comment:
As discussed, I think we should perform this check on the `CreditorsAfterOneYear` object to determine whether to create a `CurrentPeriod` in the first place. We can also remove all of the null checks - Thymeleaf handles creation of nested objects for us. But happy to defer those changes to a separate PR.

Reply:
I'll raise this as part of the 'missing tidy up' stories that were raised last week.